### DISCUSSION TITLE
How can I align input field with the buttons?

### DISCUSSION THREAD

**Discussion by nemethmik** [UI5/webcomponents#3959] [Q&A] [answered]
Greetings from a new UI5 web components fan!
I have been building a POC for a (major) mobile application using UI5 web component, Vite, lit-html and Lit Element, everything works brutally excellently. I am very happy with this stack of tools, absolutely excellent.
Here is the part from my HTML that builds the bottom part:
![paola18ui5products02](https://user-images.githubusercontent.com/38420170/133944206-12b3f1f0-dac6-43f2-bba6-8e1d58cfc973.jpg)
And here is how it looks:
![paola18ui5products01](https://user-images.githubusercontent.com/38420170/133944018-3bd013da-0478-4583-ad6a-e4bfcd7fdf22.jpg)
Here is a live version of the application [lively-field-057f5c603.azurestaticapps.net/](https://lively-field-057f5c603.azurestaticapps.net/) and here is the [repo](https://github.com/nemethmik/paola18)
Could you advise, please?
Thank you so much, Miklos

**Comment by pskelin** [comment, marked as answer]
Hi Miklos,
thanks for the nice words, really glad to hear positive experiences.

The misalignment of the components has been reported a couple of times and we plan to improve the default alignment of the components. You can track the progress here:
https://github.com/SAP/ui5-webcomponents/issues/3492

Until we fix this, you can use the suggested solution of wrapping them in a div and styling them with `dislay: flex` and aligning them vertically with `align-items: center;` as suggested in this comment and the linked sample:
https://github.com/SAP/ui5-webcomponents/issues/3492#issuecomment-880662987

It is a bit more code, but might be still necessary for a mixed case when you have other web components or even native elements that might also be misaligned.
Regards,
Peter

**Reply by nemethmik** [reply]
Thank You so much! Adding these flexbox lines is no problem, thank you for the hint, it's dead easy, I have corrected my [experimental PoC app](https://lively-field-057f5c603.azurestaticapps.net/). Miklos
